index.html

```diff
@@ -2003,23 +2042,45 @@ <h2>The <dfn>NDEFRecordType</dfn> string</h2>
       </td>
     </tr>
     <tr>
-      <td><a>External type record</a></td>
+      <td><a>Absolute-URL record</a> (TNF=3) with TYPE=<i>url</i></td>
+      <td>"`url`"</td>
+      <td>""</td>
+      <td>
+        <a>text()</a> returning <i>url</i> or<br>
+        <a>arrayBuffer()</a>
+      </td>
+    </tr>
+    <tr>
+      <td><a>External type record</a> (TNF=4) with TYPE=<a>external type</a></td>
       <td><a>external type</a></td>
       <td>"`application/octet-stream`"</td>
       <td>
         <a>text()</a> or<br>
         <a>json()</a> or<br>
-        <a>arrayBuffer()</a>
+        <a>arrayBuffer()</a> or<br>
+        <a>toRecords()</a>
       </td>
     </tr>
     <tr>
-      <td>Any other <a>NDEF record</a> type</td>
-      <td>"`opaque`"</td>
+      <td>Any embedded <a>NDEF record</a> with TYPE=<a>local type</a></td>
+      <td><a>local type</a></td>
       <td>"`application/octet-stream`"</td>
       <td>
         <a>text()</a> or<br>
         <a>json()</a> or<br>
-        <a>arrayBuffer()</a>
+        <a>arrayBuffer()</a> or<br>
+        <a>toRecords()</a>
```

Comment:
A local type record itself always supports toRecords()? i.e. its payload is a ndef message?

Reply:
I was thinking on this when added toRecords(). My reasoning was that if an app wants to create a tree deeper than 1, do we cut the tree at depth 1, or allow (provide tools) that apps could embed to infinity?

I took the route of allowing what is theoretically possible, but maybe we could limit the embedding at depth 1, saying "from here on, you can do it, but you are on your own, handle ArrayBuffer directly".

Reply:
There is no depth limit.

I can easily create an external type record and embed another external type record inside that has embedded NDEF - as well as make a local type that has NDEF.

It is not really hard to handle, you just need to measure the size and set it as payload

Reply:
OK, I agree we should not have depth limitation, so a local type record could indeed contain other records and should be able to be parsed to multiple records.